common/types/mapper/proto/admin.go

```diff
@@ -1165,21 +1167,53 @@ func ToDynamicConfigFilter(t *adminv1.DynamicConfigFilter) *types.DynamicConfigF
 
 func FromGetGlobalIsolationGroupsResponse(t *types.GetGlobalIsolationGroupsResponse) *adminv1.GetGlobalIsolationGroupsResponse {
 	if t == nil {
-		return nil
+		return &adminv1.GetGlobalIsolationGroupsResponse{}
 	}
 	cfg := isolationGroupConfigToIDL(&t.IsolationGroups)
+	if cfg == nil || len(cfg.IsolationGroups) == 0 {
+		return &adminv1.GetGlobalIsolationGroupsResponse{}
+	}
 	return &adminv1.GetGlobalIsolationGroupsResponse{
 		IsolationGroups: cfg,
 	}
 }
 
+func FromGetGlobalIsolationGroupsRequest(t *types.GetGlobalIsolationGroupsRequest) *adminv1.GetGlobalIsolationGroupsRequest {
+	if t == nil {
+		return nil
+	}
+	return &adminv1.GetGlobalIsolationGroupsRequest{}
+}
+
+func FromUpdateGlobalIsolationGroupsRequest(t *types.UpdateGlobalIsolationGroupsRequest) *adminv1.UpdateGlobalIsolationGroupsRequest {
+	if t == nil || len(t.IsolationGroups) == 0 {
```

Comment:
nil and the non-nil should be treated differently
this is just for the purpose of keeping the behavior consistent
if you search `t == nil` in this file, they are all converted to `nil`